radicle-httpd/src/raw.rs

```diff
@@ -27,12 +29,16 @@ async fn file_handler(
 ) -> impl IntoResponse {
     let storage = &profile.storage;
     let repo = Repository::open(paths::repository(storage, &project))?;
-    let blob = repo.blob(sha, &path)?;
-
     let mut response_headers = HeaderMap::new();
+
+    if repo.file(sha, &path)?.content(&repo)?.size() > MAX_BLOB_SIZE {
+        return Ok::<_, Error>((StatusCode::PAYLOAD_TOO_LARGE, response_headers, vec![]));
+    }
+
+    let blob = repo.blob(sha, &path)?;
     response_headers.insert(header::CONTENT_TYPE, "text; charset=utf-8".parse().unwrap());
```

Comment:
It might be a good idea to cover a few of the common mime types, eg. those in https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types and fallback on `text`

Reply:
`text` works as expected on both chrome and firefox. browser ignores the mime and shows text, renders images, etc and on binaries starts a download.

otherwise, I'd have to add `mime_guess` crate, which I thought wouldn't be worth increasing the tree, if doesn't add any functionality.

Reply:
Just because text works on a couple of browsers, doesn't mean it's the correct content-type..

There's no need for the `mime_guess` crate, you just have a map `file_extension -> mime_type`.